File: 345-ReverseVowelsOfAString/345-ReverseVowelsOfAString.ts
// Last updated: 4/17/2026, 12:40:22 AM
function reverseVowels(s: string): string {

    let start = 0;
    let end  =  s.length -1;
    let stringArray =  s.split("");

    const checkVowel = (c:string):boolean=>{
            return (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' || c == 'A' || c == 'E' || c == 'I' || c == 'O' || c == 'U' );

    }
    
    while(start < end)
    {
        if(!checkVowel(stringArray[start]))
        {
            start++;
            continue;
        }

                if(!checkVowel(stringArray[end]))
        {
            end--;
            continue;
        }

        let temp = stringArray[start];
        
          stringArray[start]       = stringArray[end];

          stringArray[end] =  temp;

    start++;
    end--;
    }

    return stringArray.join("");

};

File: 345-ReverseVowelsOfAString/345-ReverseVowelsOfAString.py
# Last updated: 4/17/2026, 12:40:21 AM
class Solution:
    def reverseVowels(self, s: str) -> str:

        vowels = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'}
        l = list(s)
        start, end = 0, len(l) - 1
        
        while start < end:
            while start < end and l[start] not in vowels:
                start += 1
            
            while start < end and l[end] not in vowels:
                end -= 1
            
            l[start], l[end] = l[end], l[start]
            start += 1
            end -= 1
            
        return "".join(l)


                    

        